pints/residuals_diagnostics.py

```diff
@@ -0,0 +1,272 @@
+#
+# Functions for analyzing the residuals and evaluating noise models
+#
+# This file is part of PINTS
+#  Copyright (c) 2017-2019, University of Oxford.
+#  For licensing informating, see the LICENSE file distributed with the PINTS
+#  software package.
+#
+from __future__ import absolute_import, division
+import math
+
+
+def plot_residuals_autocorrelation(parameters,
+                                   problem,
+                                   max_lag=10,
+                                   thinning=None,
+                                   significance_level=0.05,
+                                   posterior_interval=0.95):
+    r"""
+    Generate an autocorrelation plot of the residuals.
+
+    This function can be used to analyze the results of either optimisation or
+    MCMC Bayesian inference. When multiple samples of the residuals are present
+    (corresponding to multiple MCMC samples), the plot illustrates the
+    distribution of autocorrelations across the MCMC samples. At each lag,
+    a point is drawn at the median autocorrelation, and a line is drawn giving
+    the percentile range of the posterior interval specified as an argument (by
+    default, the 2.5th to the 97.5th percentile).
+
+    When multiple outputs are present, one residuals plot will be generated for
+    each output.
+
+    When a significance level is provided, confidence bounds for the sample
+    autocorrelations under the assumption of IID residuals are drawn on the
+    plot. Many of the observed residuals autocorrelations falling outside these
+    bounds could imply evidence against the residuals being IID.
+
+    Under the assumption that the residuals of length :math:`n` are IID with
+    mean 0 and variance :math:`\sigma^2`, for large :math:`n` the residuals
+    sample autocorrelations are approximately IID Normal(mean=0, variance=1/n).
+    This result is proved in [1]_ (see Theorem 7.2.2 and Example 7.2.1).
+    Therefore, the confidence bounds can be calculated by :math:`\pm z^*
+    n^{-1/2}` for the appropriate critical value :math:`z^*`.
+
+    This function returns a ``matplotlib`` figure.
+
+    Parameters
+    ----------
+    parameters
+        The parameter values with shape ``(n_samples, n_parameters)``. When
+        passing a single best fit parameter vector, ``n_samples`` will be 1.
+    problem
+        The problem given by a :class:``pints.SingleOutputProblem`` or
```

Comment:
For mysterious reasons, the authors of restructuredtext/sphinx decided that it should be:

```
Reference to ``a_variable`` or ``1 + general_code()`` (double ticks)
A reference to a class or method (gets converted to a hyperlink) :meth:`a_single_tick()`
```